pkg/azurefile/controllerserver.go

```diff
@@ -1205,3 +1232,28 @@ func isValidVolumeCapabilities(volCaps []*csi.VolumeCapability) error {
 	}
 	return nil
 }
+
+func generateSASToken(accountName, accountKey, storageEndpointSuffix string) (string, error) {
+	credential, err := service.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	serviceClient, err := service.NewClientWithSharedKeyCredential(fmt.Sprintf("https://%s.file.%s/", accountName, storageEndpointSuffix), credential, nil)
+	if err != nil {
+		return "", status.Errorf(codes.Internal, fmt.Sprintf("failed to generate sas token in creating new client with shared key credential, accountName: %s, accountKey: %s, err: %s", accountName, accountKey, err.Error()))
+	}
+	nowTime := time.Now()
+	sasURL, err := serviceClient.GetSASURL(
+		sas.AccountResourceTypes{Object: true, Service: true, Container: true},
+		sas.AccountPermissions{Read: true, List: true, Write: true},
+		time.Now().Add(4*time.Hour), &service.GetSASURLOptions{StartTime: &nowTime})
```

Comment:
`4*time.Hour` is too long, pls do some perf test to verify what's the appropiate time cost. And also add a feature flag in the driver to control this

Reply:
I use a `sasExpiryTime` in driver option and set 30(30 minutes) for default.